authz/providers/azure/rbac/rbac.go

```diff
@@ -375,7 +378,94 @@ func (a *AccessInfo) CheckAccess(request *authzv1.SubjectAccessReviewSpec) (*aut
 
 		finalStatus = status
 	}
-	return finalStatus, nil
+	if finalStatus != nil && finalStatus.Allowed {
+		klog.V(5).Infof("Checkaccess request is allowed for user %s", checkAccessUsername)
+		return finalStatus, nil
+	}
+
+	if !a.enableManagedNamespaceRBAC || a.clusterType != managedClusters {
+		klog.V(5).Infof("Checkaccess request is denied for user %s", checkAccessUsername)
+		return finalStatus, nil
+	}
+
+	klog.V(5).Infof("Falling back to checking managed namespace scope", checkAccessUsername)
+
+	checkAccessURLManagedNS := *a.apiURL
```

Comment:
this point on, pretty much a duplicate of the above to not break existing scopes

Reply:
can we refactor them into function? my concern is if we will change the code in one place and miss in the other.

Reply:
the code itself is already complicated. duplicating that will make people scratch their head even harder

Reply:
refactored